File: algo/dynamic_programming/canSum.c
#include<stdio.h>

int canSum(int n,int *numbers,int len)
{
	if(n == 0) 
		return 1;
	if(n < 0)
		return 0;
	int i = 0;
	while(i < len)
	{
		if(canSum(n - numbers[i],numbers,len) == 1)
			return 1;
		i++;
	}
	return 0;

}	

int main()
{
	int numbers[3] = {3,5,11};
	int len = 3;
	int x = canSum(8,numbers,len);
	printf("%d\n",x);
}

File: algo/dynamic_programming/canSum.js
const canSum = (n,numbers) => {
	if(n === 0) return true;
	if(n < 0) return false;
	for(let i of numbers)
	{
		const remainder = n - i;
		if(canSum(remainder,numbers) === true)
		{
			return true;
		}
	}
	return false;
}

console.log(canSum(7,[1,2]));
console.log(canSum(7,[4,10]));
console.log(canSum(8,[3,5,11]));
